### ISSUE TITLE
Keep 'Content type' when blob is replaced

### ISSUE THREAD

**Issue by hmaland** [Microsoft/AzureStorageExplorer#1208]
**Is your feature request related to a problem? Please describe.**
When we upload markdown (*.md) file as blob, the ContentType is set to 'text/x-markdown'. In order to display correct encoding when viewed through a browser, we need to update the blob property ContentType to 'text/x-markdown; charset=utf-8'. When this blob is updated with a new version (replaced) the ContentType is set back to 'text/x-markdown'.

**Describe the solution you'd like**
Keep custom edited ContentType.

**Describe alternatives you've considered**
As an alternative, we could at least get a warning that the ContentType will be updated from A to B after the replacement.

**Additional context**
Testet in Azure Storage Explorer 1.6.2

**Comment by MRayermannMSFT** [comment]
@hmaland , is there a reason you aren't using the content encoding property for setting the charset?

**Comment by hmaland** [comment]
Setting content encoding property to 'utf-8' doesn't seem to help when viewed through a browser.

**Comment by MRayermannMSFT** [comment]
What do you mean by viewed through a browser? Are you trying to load the markdown file in your browser directly? Like copy and pasting a URL to it? Trying to understand so we can reproduce the issue and figure out what is the "right" thing for us/you to be doing. Thanks. 😄

**Comment by hmaland** [comment]
Yes, that is exactly what I mean. Just pasting the URL to the markdown file into Chrome opens the file for viewing inside the browser (just as plain text). International characters are not displayed correctly since the browser does not know anything about the charset. So we tested that by setting the ContentType property to 'text/x-markdown; charset=utf-8', the file content was displayed correctly in Chrome. 

(This is not the main usage scenario for this blob-file though, it will normally be downloaded by javascript code and displayed with real markdown formatting.)

**Comment by MRayermannMSFT** [comment]
Ok, I'm going to try this out later today/next week. I also need to some more reading on content types. Will let you know. :) Thanks for the patience.

**Comment by MRayermannMSFT** [comment]
Hey @hmaland , so a few things I'll say after testing around:

1. This would also annoy the heck out of me, so I agree we need to solve this problem
2. I can definitely repro Chrome not knowing how to display unicode/intl. characters if the Content Type does not include the charset.
3. If you want to for now, you can enable the Preview AzCopy feature. It correctly uses a Content Type that includes a charset. But, what it doesn't do is use `text/x-markdown`, instead it uses `text/plain`. But, Chrome has no issue rendering the file. Hopefully your JS code also doesn't care if it's `text/x-markdown` vs `text/plain`.
![image](https://user-images.githubusercontent.com/21112697/54059893-9ec05d80-41af-11e9-903c-bbe065d49032.png)

Anyways, I'm going to email both the JS library team and the AzCopy team to see if we can get them both to do what appears to be the right thing, just leaving the ContentType alone.

**Comment by MRayermannMSFT** [comment]
Hi @hmaland , I have some more updates for you after talking to those teams.

1. At this time, the JS upload is not going to be able to do the charset addition to your content type. The library the Storage SDK is using to detect content type of a file doesn't have that feature.
2. AzCopy on the other hand (as we've seen) is able to figure out the charset! Aditionally, if having the beginning of the content type being text/x-markdown vs text/plain is important to you, then you can make that happen.
- If you're on a Unix OS, your OS should have a mime.types file, common loactions for it are `/etc/mime.types`, `/etc/apache2/mime.types`, `/etc/apache/mime.types`
- If you're on Windows, you should be able to add the correct mime type to the Windows registry
A web search should get you some more details on adding the mime.type correctly for your OS.

If you're able to do those successfully, AzCopy v10 should apparently respect the change.

**Comment by MRayermannMSFT** [comment]
Moving to future...

**Comment by StingyJack** [comment]
This has started happening with json files we have stored as block blob.  They exist as content type `application/json`, and when we replace them they are now getting set as content type `text/plain; charset=utf-8`. 

Its not a blocker but is unexpected. We can edit the content type in the properties dialog for the item if it does become a blocker..

**Comment by hmaland** [comment]
The original behavior I reported is no longer happening when we upload/replace an existing blob. I'm not sure in which version of Storage Explorer this was fixed, but the reported issue can be closed for my part.
